### ISSUE TITLE
Implicit short names not considered when deciding if version/help have short names

### ISSUE THREAD

**Issue by LennartH** [scallop/scallop#177]
It seems like the implicit short names of options aren't considered when the `help` and `version` command are added to the config.

Consider the following minimal example:
```scala
object Main {

  class Config(args: Seq[String]) extends ScallopConf(args) {
    // Setting the short name explicitly works as expected
    val verbose = opt[Boolean]()
    val hard = opt[Boolean]()

    version("Version 1.0") // It doesn't matter if version is called before or after the options

    verify()
  }

  def main(args: Array[String]): Unit = {
    val config = new Config(args)
    println(config.summary)
  }
}
```
Executing with `-h` or `-v` doesn't print the the configs summary, but the help or version string. The help string looks like this, suggesting that the short names would be valid for the options:
```
Version 1.0
  -h, --hard
  -v, --verbose
      --help      Show help message
      --version   Show version of this program
```
Executing with `-hv` on the other hand, sets both options to `true`.

A similar issue occurs when a subcommand has an option with `v` as short name:
```scala
class ConfigWithSubcommand(args: Seq[String]) extends ScallopConf(args) {

  val cmd = new Subcommand("cmd") {
    val vopt = opt[Boolean](short = 'v')
  }
  addSubcommand(cmd)

  version("Version 1.0")

  verify()
}
```
Executing with `cmd -v` prints the version, despite the explicitly specified short name and the option being set after the command name (unless the root config has an option with an explicit short name `v` or doesn't specify a version string). The help string again suggests that `-v` is valid to set the option.

**Comment by Rogach** [comment]
Yes, that's a bug, will fix. Thanks for catching it!

**Comment by Rogach** [comment]
Released `3.1.4` with the fix, should sync to Maven Central in about an hour.

**Comment by LennartH** [comment]
Awesome! Thanks for the fast response!

But the issue with the subcommand remains. The following snippet prints the version instead of the summary:
```scala
val config = new ScallopConf(Seq("cmd", "-v")) {
  version("Version 1.0")
  val cmd = new Subcommand("cmd") {
    val verbose = opt[Boolean]()
  }
  addSubcommand(cmd)
  verify()
}
println(config.summary)
```

**Comment by Rogach** [comment]
Released `3.1.5` with the fix, please test the new version.

**Comment by LennartH** [comment]
It's working now. Thanks for the fix!
